### ISSUE TITLE
False positive PropertyNotSetInConstructor

### ISSUE THREAD

**Issue by viktorprogger** [vimeo/psalm#7755]
Hi! My code is working ok at psalm.dev, but in my project I've got this issue in every class with a such block of code:
![image](https://user-images.githubusercontent.com/7670669/156877605-289920f9-0cf1-4073-9cb0-dba2bb04b661.png)
![image](https://user-images.githubusercontent.com/7670669/156877616-32f5e34c-34bb-40fa-b1a0-58d6b0a319a1.png)

Also I can't use DIP with DI container in this place due to some restrictions of the class instantiation.

I tried this snippet at psalm.dev, but it has no errors;
```php
final class Foo {
    private string $activitySettings;
    
    public function __construct()
    {
        /** 
         * @var \Psr\Container\ContainerInterface $serviceLocator
         * @var string $activitySettings
         */
        $activitySettings = $serviceLocator->get('bar');
        $this->activitySettings = $activitySettings;
    }
}
```

Psalm version I've used is 4.21.0.

**Comment by orklah** [comment]
Do you have any other error in this file? Including suppressed or baselined errors?

**Comment by AndrolGenhald** [comment]
You're missing a `$` at `/** @var ActivitySettings activitySettings */`, but it shouldn't cause `PropertyNotSetInConstructor`. Can you try Psalm 4.22.0 and master and see if they show the issue as well?

**Comment by viktorprogger** [comment]
@orklah, there are no other psalm issues in the file.
@AndrolGenhald, yeap, it reproduces with 4.22.0 and dev-master.

**Comment by AndrolGenhald** [comment]
Do you by any chance have `DispatcherWorkflow` stubbed somewhere? Are you using a Psalm plugin? A small repository that reproduces the problem would help a lot.

**Comment by viktorprogger** [comment]
No stubs and no plugins. And I still don't know how to reproduce this behavior outside of the project. I'll try to create a reproducible sample once again this week.
